app/services/forest_liana/ability/permission/smart_action_approval.rb

```diff
@@ -0,0 +1,67 @@
+require 'jwt'
+
+module ForestLiana
+  module Ability
+    module Permission
+    class SmartActionApproval
+
+      def initialize(parameters, collection, smart_action, user)
+        @parameters = parameters
+        @collection = collection
+        @smart_action = smart_action
+        @user = user
+      end
+
+      def can_trigger?
```

Comment:
```suggestion
      def can_execute?
```
As it checks trigger and approve?

Reply:
good idea 👍